freshmaker/pyxis_gql_async.py

```diff
@@ -0,0 +1,472 @@
+            ds.ContainerRepository.registry,
+            ds.ContainerRepository.repository,
+        ]
+        return projection
+
+    def _get_image_projection(self, include_rpms: Optional[bool] = True) -> list[DSLField]:
+        ds = self.dsl_schema
+        projection = [
+            ds.ContainerImage.architecture,
+            ds.ContainerImage.brew.select(
+                ds.Brew.build,
+            ),
+            ds.ContainerImage.content_sets,
+            ds.ContainerImage.parent_brew_build,
+            ds.ContainerImage.parsed_data.select(
+                ds.ParsedData.labels.select(
+                    ds.Label.name,
+                    ds.Label.value,
+                ),
+            ),
+            ds.ContainerImage.repositories.select(
+                ds.ContainerImageRepo.registry,
+                ds.ContainerImageRepo.repository,
+                ds.ContainerImageRepo.published,
+                ds.ContainerImageRepo.tags.select(
+                    ds.ContainerImageRepoTag.name,
+                ),
+            ),
+        ]
+
+        # Include rpm manifest data in result, use edges to get the rpm manifest
+        # data because the direct rpm manifest field doesn't include all data
+        if include_rpms:
+            projection.append(
+                ds.ContainerImage.edges.select(
+                    ds.ContainerImageEdges.rpm_manifest.select(
+                        ds.ContainerImageRPMManifestResponse.data.select(
+                            ds.ContainerImageRPMManifest.image_id,
+                            ds.ContainerImageRPMManifest.rpms.select(
+                                ds.RpmsItems.name,
+                                ds.RpmsItems.nvra,
+                                ds.RpmsItems.srpm_name,
+                                ds.RpmsItems.srpm_nevra,
+                            ),
+                        ),
+                    ),
+                )
+            )
+        return projection
+
+    async def find_repositories(
+        self,
+        published: Optional[bool] = None,
+        release_categories: Optional[list[str]] = None,
+        auto_rebuild_tags: Optional[list[str]] = None,
+    ) -> list[dict[str, Any]]:
+        """Get image repositories
+
+        :param bool published: published or unpublished repositories
+        :param list release_categories: list of release categories
+        :param list auto_rebuild_tags: list of tags enabled for auto rebuild
+        :return: list of image repositories
+        :rtype: list
+        """
+        query_filter: dict = {}
+        query_filter["and"] = []
+        # Query Red Hat repositories only
+        query_filter["and"].append({"vendor_label": {"eq": "redhat"}})
+
+        if published is not None and isinstance(published, bool):
+            query_filter["and"].append(
+                {"published": {"eq": True}}
+            )  # TODO what if published=False? should True be still appended?
```

Comment:
This has been fixed in `pyxis` branch, some modules in `main` branch are outdated.

Reply:
I see. Yeah, I definitely missed out quite a bit by not getting the code from the `pyxis` branch 😅